test/promql/tests/test_notready_pods.yaml

```diff
@@ -0,0 +1,162 @@
+      - eval_time: 15m
+        alertname: PodsNotReady
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: ns-1
+              pod: pod-1
+              phase: Pending
+            exp_annotations:
+              summary: "Pod pod-1 is not ready"
+              description: "Pod pod-1 in namespace ns-1 is not ready for more than 15 minutes."
+
+  - interval: 1m
+    input_series:
+      # Pod is in the Unknown state and scheduled, so it will be alerted.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-2", phase="Pending"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-2", phase="Unknown"}'
+        values: '1x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-2", phase="Failed"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-2", phase="Running"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-2", phase="Succeeded"}'
+        values: '0x14'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-2", pod="pod-1"}'
+        values: '0x14'
+
+    alert_rule_test:
+      - eval_time: 15m
+        alertname: PodsNotReady
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: ns-2
+              pod: pod-1
+              phase: Unknown
+            exp_annotations:
+              summary: "Pod pod-1 is not ready"
+              description: "Pod pod-1 in namespace ns-2 is not ready for more than 15 minutes."
+
+  - interval: 1m
+    input_series:
+      # Pod is in the Failed state and scheduled, so it will be alerted.
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-3", phase="Pending"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-3", phase="Unknown"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-3", phase="Failed"}'
+        values: '1x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-3", phase="Running"}'
+        values: '0x14'
+      - series: 'kube_pod_status_phase{pod="pod-1", namespace="ns-3", phase="Succeeded"}'
+        values: '0x14'
+      - series: 'kube_pod_status_unschedulable{namespace="ns-3", pod="pod-1"}'
+        values: '0x14'
+
+    alert_rule_test:
+      - eval_time: 15m
+        alertname: PodsNotReady
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              namespace: ns-3
+              pod: pod-1
+              phase: Failed
+            exp_annotations:
+              summary: "Pod pod-1 is not ready"
+              description: "Pod pod-1 in namespace ns-3 is not ready for more than 15 minutes."
+
+  - interval: 1m
+    input_series:
+      # Not Alerted cases:
+      # Pod is in the Unknown state and unscheduled (unscheduled once in the 15 min interval), so it will not be alerted.
```

Comment:
```suggestion
      # Pod is in the Unknown state and unscheduled (unscheduled once in the 15 min
      # interval), so it will not be alerted.
```